kernel/src/engine/arrow_expression/evaluate_expression.rs

```diff
@@ -118,6 +120,28 @@ pub fn evaluate_expression(
         (Predicate(_), Some(data_type)) => Err(Error::generic(format!(
             "Predicate evaluation produces boolean output, but caller expects {data_type:?}"
         ))),
+        (Unary(UnaryExpression { op, expr }), data_type) => {
+            match op {
+                ToJson => {
+                    if let Some(data_type) = data_type {
+                        if data_type != &DataType::STRING {
+                            return Err(Error::generic(format!(
+                                "ToJson operator requires STRING output, but got {data_type:?}"
+                            )));
+                        }
+                    }
+                }
+            };
```

Comment:
I added this "additional" match block so that we can fail before evaluating the input expression if the desired output data type is invalid.